src/test/java/org/painandsuffer/BattleTest.java

```diff
@@ -0,0 +1,147 @@
+package org.painandsuffer;
+
+import org.junit.jupiter.api.Test;
+import org.painandsuffer.battle.Battle;
+import org.painandsuffer.characters.Creature;
+import org.painandsuffer.characters.adventurers.Mage;
+import org.painandsuffer.characters.adventurers.Rogue;
+import org.painandsuffer.characters.adventurers.Warrior;
+import org.painandsuffer.items.armour.ArmourSet;
+import org.painandsuffer.items.armour.chest.Chainmail;
+import org.painandsuffer.items.armour.chest.LeatherArmour;
+import org.painandsuffer.items.armour.chest.Robe;
+import org.painandsuffer.items.armour.gloves.ChainGloves;
+import org.painandsuffer.items.armour.gloves.LeatherGloves;
+import org.painandsuffer.items.weapons.Dagger;
+import org.painandsuffer.items.weapons.Staff;
+import org.painandsuffer.items.weapons.Sword;
+
+import static org.junit.jupiter.api.Assertions.*;
+
```

Comment:
General considerations:
1. Clearly delineate what constitutes unit vs. integration testing within your test suite. Unit tests should focus on individual methods or small units of logic within the Battle class, assuming dependencies behave correctly. Integration tests should focus on how the Battle class integrates with Creature subclasses and other game components.
2. Even within integration tests, consider mocking external dependencies if they are not central to the test scenario. For example, you might mock a logging component or an external service, but you would likely use actual instances of Creature subclasses and their equipment.
3. Ensure tests are focused on the behavior and outcomes of the battle scenarios rather than the internal implementation details of the Battle class. Verify the end state of creatures post-battle, the correctness of the round counter, and any other significant side effects of the battle logic.
4. Combine similar tests or convert them into parameterized tests. For example, the tests for checking health reduction can be parameterized to test different weapons or armor effects within a single test method, thus focusing on how these factors influence the battle outcome.